src/sonic-config-engine/minigraph.py

```diff
@@ -157,10 +190,19 @@ def parse_png(png, hname):
                             elif node.tag == str(QName(ns, "EndDevice")):
                                 mgmt_dev = node.text
 
-    return (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speeds, console_ports)
+    return (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speeds, console_ports, device_neighbors)
+
 
+def parse_dpg(dpg, hname, device_hostname):
 
-def parse_dpg(dpg, hname):
+    #In Multi-NPU platforms the acl intfs are defined for only for the host.
+    # to get the aclintfs node first.
+    for child in dpg:
+
+        aclintfs = child.find(str(QName(ns, "AclInterfaces")))
+        if aclintfs is not None:
+            break 
```

Comment:
i do not understand the change here. why do we need two for loop now? if there is no aclinterfaces, then skip aclinterface processing. we can still do it in one pass.

Reply:
In the Multi-asic minigraph, the aclIntf is present for the complete host not per asic, so while parsing to generate the asic configuation, we need to get the aclIntf node first.

Fixed to remove extra loop